test/dark-gravity-wave-test.js

```diff
@@ -1,126 +1,286 @@
+    target: 0x1b17e949,
+    timestamp: 1438531015,
 }, {
-  height: 312684,
-  target: 0x1b1797b7,
-  timestamp: 1438531231,
+    height: 312684,
+    target: 0x1b1797b7,
+    timestamp: 1438531231,
 }, {
-  height: 312685,
-  target: 0x1b19406d,
-  timestamp: 1438531237,
+    height: 312685,
+    target: 0x1b19406d,
+    timestamp: 1438531237,
 }, {
-  height: 312686,
-  target: 0x1b193835,
-  timestamp: 1438531418,
+    height: 312686,
+    target: 0x1b193835,
+    timestamp: 1438531418,
 }, {
-  height: 312687,
-  target: 0x1b15f370,
-  timestamp: 1438531477,
+    height: 312687,
+    target: 0x1b15f370,
+    timestamp: 1438531477,
 }, {
-  height: 312688,
-  target: 0x1b15bda6,
-  timestamp: 1438531515,
+    height: 312688,
+    target: 0x1b15bda6,
+    timestamp: 1438531515,
 }, {
-  height: 312689,
-  target: 0x1b15792d,
-  timestamp: 1438531838,
+    height: 312689,
+    target: 0x1b15792d,
+    timestamp: 1438531838,
 }, {
-  height: 312690,
-  target: 0x1b16dca7,
-  timestamp: 1438531999,
+    height: 312690,
+    target: 0x1b16dca7,//454483111
+    timestamp: 1438531999,
 }, {
-  height: 312691,
-  target: 0x1b1777d4,
-  timestamp: 1438531991,
+    height: 312691,
+    target: 0x1b1777d4,//454522836
+    timestamp: 1438531991,
 }];
 
 describe('dark gravity wave', () => {
-  describe('difficulty calculation', () => {
-    it('should be valid for low enough target', () => {
-      const highTargetBits = 0x1b0777d4;
-      const result = dgw.isValidTarget(highTargetBits, blocks);
-      expect(result).to.equal(true);
-    });
+    describe('difficulty calculation', () => {
+        it('should be valid for low enough target', () => {
+            const highTargetBits = 0x1b0777d4; // 453474260
+            const result = dgw.isValidTarget(highTargetBits, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be invalid for too high target', () => {
+            const highTargetBits = 0x208fffff; // 546308095
+            const result = dgw.isValidTarget(highTargetBits, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid for the max regtest target if less than 25 blocks has been mined', () => {
+            const highTargetBits = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetBits, blocks.slice(0, 10));
+            expect(result).to.equal(true);
+        });
+
+                it('should be valid for int headerBits', () => {
+            const highTargetInt = 1000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be invalid maxTarget = 0x207fffff', () => {
+            const highTargetInt = 0x207fffff;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid when num blocks < maxBlocks and maxTarget = 0x207fffff', () => {//TODO is it True?
```

Comment:
yes as I understand if less than 25 blocks has been mined the maxtarget is valid.